analysis/topEFT/dataDrivenEstimation.py

```diff
@@ -0,0 +1,106 @@
+import argparse
+from coffea import hist, processor
+from topcoffea.modules.YieldTools import YieldTools
+from topcoffea.modules.GetValuesFromJsons import get_lumi
+import cloudpickle
+
+class DataDrivenProducer: 
+    def __init__(self, inputHist, outputName, year, doDDFakes=True, doDDFlips=False):
+        self.yt=YieldTools()
+        if type(inputHist) == str: # we are plugging a pickle file
+            self.inhist=yt.get_hist_from_pkl(inputHist)
+        else: # we already have the histogram
+            self.inhist=inputHist
+        self.outputName=outputName
+        if doDDFlips: 
+            raise RuntimeError("Data driven flips not yet implemented")
+        self.doDDFlips=doDDFlips
+        self.doDDFakes=doDDFakes
+        self.dataName='data'
+        self.outHist=None
+        self.year=year
+
+        self.DDFakes()
+
+    def DDFakes(self):
+        if self.outHist!=None:  # already some processing has been done, so using what is available
+            self.inhist=self.outHist
+            
+        self.outHist={}
+        for key,histo in self.inhist.items():
+            if key == 'SumOfEFTweights': 
+                # we will deal later with the sum of weights (to do) 
+                continue 
+
+            if not len(histo.values()): # histo is empty, so we just integrate over appl and keep an empty histo
+                self.outHist[key]=histo.integrate('appl')
+                continue
+
+            if not self.doDDFakes:
+                # if we are not gonna use the data-driven, then we don't care about the application region, so we get rid of it, and the associated dimension
+                srs=[ histo.integrate('appl',ident) for ident in histo.identifiers('appl') if 'SR' in ident.name]
+                if not len(srs):
+                    continue # to do, this should be a runtime errror
```

Comment:
I agree with the comment. I'll just put this review comment here so we remember to add `raise Exception()`.

Reply:
sure, will do